ddtrace/contrib/botocore/patch.py

```diff
@@ -519,32 +519,17 @@ def patched_api_call(original_func, instance, args, kwargs):
                 try:
                     if "Messages" in result:
                         for message in result["Messages"]:
+                            pathway = None
                             if (
                                 "MessageAttributes" in message
                                 and "_datadog" in message["MessageAttributes"]
                                 and "StringValue" in message["MessageAttributes"]["_datadog"]
                             ):
-                                if PROPAGATION_KEY_BASE_64 in json.loads(
-                                    message["MessageAttributes"]["_datadog"]["StringValue"]
-                                ):
-                                    pathway = json.loads(message["MessageAttributes"]["_datadog"]["StringValue"])[
-                                        PROPAGATION_KEY_BASE_64
-                                    ]
-                                    ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
-                                    ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
-
-                                else:
-                                    log.debug(
-                                        "Unable to find data streams context for the message. Data "
-                                        "streams monitoring was not enabled by the application "
-                                        "sending the message at the time of the send."
-                                    )
-                            else:
-                                log.debug(
-                                    "Unable to find trace context for the message. Please ensure that your Datadog "
-                                    "agent is correctly configured on the application sending the message and that "
-                                    "the SQS messages being sent have 9 or less Message Attributes."
+                                pathway = json.loads(message["MessageAttributes"]["_datadog"]["StringValue"]).get(
+                                    PROPAGATION_KEY_BASE_64, None
                                 )
+                            ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
+                            ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
```

Comment:
We should avoid calling data.encode("utf-8") when data is None. Handling an AttributeError here is less than ideal.

Instead we can just create a new pathway directly. With this suggestion we can revert handling the AttributeError in decode_pathway_b64()

```suggestion
                            if pathway is not None:
                               ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
                            else:
                               ctx = pin.tracer.data_streams_processor.new_pathway()
                            ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
```